src/mt_table.jl

```diff
@@ -0,0 +1,62 @@
+import Base: setindex!, haskey, getindex
+
+struct TableBlock
+  depth::Int
+  entry::Dict{String, Union{TableBlock, Some}}
```

Comment:
```suggestion
  entry::Dict{Symbol, Union{TableBlock, Some}}
```
這樣會不會好點？

Reply:
I need to split the key with `'.'`, `Symbol` will require many conversion between `Symbol` and `String`

Reply:
Isn't it just one module/subfolder one `TableBlock`? Or I just missing something?

Reply:
there are multi-level of `TableBlock`, the depth represented by each `'.'` in the key.

Reply:
So, what is your design here? I think you have the nested `TableBlock` so that you can just break module name into pieces and you don't need a `.` anymore.

Reply:
Then you will be passing a `Vector{SubString}` around, but we don't really need to find all the `'.'` in the string if there are no entry of the some prefix. breaking the `'.'` at each level help early stoping.